### PR TITLE
fix: :hammer: update Dockerfile and Docker -compose for local dev

### PR DESCRIPTION
**Pull request by LuisBlanche**
Some requirements were missing from the `Dockerfile` after migrating to hatch 

- Also added more logs on the entrypoint for debugging 
- Also added depends on pstgres for carbonserver to wait before starting

### CODE REVIEW

**Comment by inimaz** [comment]
Looks good to me! To test it:
```docker compose up--build -d```

Go to `localhost:8008/docs` to see the FastApi page

File: docker-compose.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ services:
     networks:
       - codecarbon_net
 
-  postgres:
+  postgres:
```
Comment (benoit-cty) [inline docker-compose.yml:31]:
```suggestion
  postgres:
```

File: carbonserver/docker/Dockerfile [hunk 2]
```diff
@@ -1,31 +1,29 @@
...
-# For production, you could use python:3.8-slim
 FROM ubuntu:22.04
 
-# set work directory
 WORKDIR /carbonserver
 
-# set env variables
 ENV PYTHONDONTWRITEBYTECODE 1
 ENV PYTHONUNBUFFERED 1
-# Prevent apt to ask for region
```
Comment (benoit-cty) [inline carbonserver/docker/Dockerfile:13]: I would have kept the comment.

File: carbonserver/docker/Dockerfile [hunk 3]
```diff
@@ -1,31 +1,29 @@
-# Dockerfile
-
-# Use Ubuntu to install 3 versions of Python for testing
```
Comment (benoit-cty) [inline carbonserver/docker/Dockerfile:3]: I would have kept the comment.

**Review by benoit-cty — APPROVED** [review]
Thanks for the fix.

I just have some no-blocking comments.

**Pull request merged**